stream_pager.go

```diff
@@ -223,6 +231,7 @@ func (p *StreamPager) NextMsg(ctx context.Context) (msg *nats.Msg, last bool, er
 		}
 	}
 
+filterNextDirectly:
```

Comment:
Wont this still have the effect of sending to the client less than a page full of messages when some are filtered?

This method would need significant rewriting because it fetches a batch equal to page size, so when we skip one we also need to fetch one so that there is a page full of messages sent when we tell the client we sent a page full.

If the client says he wants 25 messages in a page he needs to get 25 even when 5 were filtered, not 20 and then told its a end of page.